src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -399,6 +406,198 @@ private void WriteMethods(List<MethodBuilderImpl> methods, List<GenericTypeParam
+                constantList: MetadataTokens.LocalConstantHandle(1), scope._startOffset, scope._endOffset - scope._startOffset);
+
+            if (scope._children != null)
+            {
+                foreach (Scope childScope in scope._children)
+                {
+                    AddLocalScope(methodHandle, parentImport, MetadataTokens.LocalVariableHandle(_pdbMetadata.GetRowCount(TableIndex.LocalVariable) + 1), childScope);
+                }
+            }
+        }
+
+        private LocalVariableHandle GetLocalVariableHandle(List<LocalBuilder>? locals, LocalVariableHandle firstLocalOfLastScope)
+        {
+            if (locals != null)
+            {
+                bool firstLocalSet = false;
+                foreach (LocalBuilderImpl local in locals)
+                {
+                    if (!string.IsNullOrEmpty(local.Name))
+                    {
+                        LocalVariableHandle localHandle = _pdbMetadata.AddLocalVariable(LocalVariableAttributes.None, local.LocalIndex,
+                                                          local.Name == null ? _pdbMetadata.GetOrAddString(string.Empty) : _pdbMetadata.GetOrAddString(local.Name));
+                        if (!firstLocalSet)
+                        {
+                            firstLocalOfLastScope = localHandle;
+                            firstLocalSet = true;
+                        }
+                    }
+                }
+            }
+
+            return firstLocalOfLastScope;
+        }
+
+        private ImportScopeHandle GetImportScopeHandle(List<string>? importNamespaces, ImportScopeHandle parent)
+        {
+            if (importNamespaces == null)
+            {
+                return default;
+            }
+
+            BlobBuilder importBlob = new BlobBuilder();
+
+            foreach (string importNs in importNamespaces)
+            {
+                importBlob.WriteByte((byte)ImportDefinitionKind.ImportNamespace);
+                importBlob.WriteCompressedInteger(MetadataTokens.GetHeapOffset(_pdbMetadata.GetOrAddBlobUTF8(importNs)));
+            }
+
+            return _pdbMetadata.AddImportScope(parent, _pdbMetadata.GetOrAddBlob(importBlob));
+        }
+
+        private static void SerializeDeltaLinesAndColumns(BlobBuilder spBuilder, SequencePoint sequencePoint)
+        {
+            int deltaLines = sequencePoint.EndLine - sequencePoint.StartLine;
+            int deltaColumns = sequencePoint.EndColumn - sequencePoint.StartColumn;
+
+            // only hidden sequence points have zero width
+            Debug.Assert(deltaLines != 0 || deltaColumns != 0 || sequencePoint.IsHidden);
+
+            spBuilder.WriteCompressedInteger(deltaLines);
+
+            if (deltaLines == 0)
+            {
+                spBuilder.WriteCompressedInteger(deltaColumns);
+            }
+            else
+            {
+                spBuilder.WriteCompressedSignedInteger(deltaColumns);
+            }
```

Comment:
```suggestion
            {
                Debug.Assert(deltaLines > 0);
                spBuilder.WriteCompressedSignedInteger(deltaColumns);
            }
```

The specification doesn't define the state if `deltaLines` is < 0.